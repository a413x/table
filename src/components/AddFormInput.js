import React, { Component } from "react";
import PropTypes from 'prop-types';

import validate from '../utils/validate.js';

class AddFormInput extends Component{
  inputChange(e,field,type){
    const val = type === 'number' ? parseInt(e.target.value) : e.target.value;
    let isValid = false;
    if(validate.validateFunc(validate.regExs[type],val)){
      isValid = true;
      this.props.addingUserChange(field,val);
    }
    this.props.validationChange(this.props.index,isValid);
  }
  render(){
    const col = this.props.col;
    const valid = this.props.valid ? 'is-valid' : 'is-invalid';
    return(
      <div>
        <label>{col.title}</label>
        <input className = {'form-control ' + valid}
          onChange = {(e) => this.inputChange(e,col.field,col.type)} />
        <div className="invalid-feedback">
          Введите корректное значение.
        </div>
      </div>
    );
  }
}

AddFormInput.propTypes = {
  index: PropTypes.number.isRequired,
  col: PropTypes.object.isRequired,
  valid: PropTypes.bool.isRequired,
  addingUserChange: PropTypes.func.isRequired,
  validationChange: PropTypes.func.isRequired
};

export default AddFormInput;
