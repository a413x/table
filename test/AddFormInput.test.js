import React from "react";
import {shallow,mount} from 'enzyme';

import AddFormInput from "../src/components/AddFormInput";
import data from './data.js';

describe('tests for AddFormInput component',() => {
  const tableColumns = data.tableColumns;
  const mockAddingUserChange = jest.fn();
  const mockValidationChange = jest.fn();
  const props = {
    index: 0,
    col: tableColumns[0],
    valid: false,
    addingUserChange: mockAddingUserChange,
    validationChange: mockValidationChange
  };

  const addInputContainer = shallow(<AddFormInput {...props} />);

  it('renders properly',()=>{
    expect(addInputContainer).toMatchSnapshot();
  });

  it('renders properly with valid prop = true',()=>{
    let nextProps = props;
    nextProps.valid = true;
    expect(addInputContainer.find('input').hasClass('is-invalid'));
    const nextAddInputContainer = shallow(<AddFormInput {...nextProps} />);
    expect(nextAddInputContainer.find('input').hasClass('is-valid'));
  });

  describe('input change called',()=>{
    it('with invalid value',()=>{
      addInputContainer.find('input').simulate('change',{
        target:{value:''}
      });
      expect(mockAddingUserChange).toHaveBeenCalledTimes(0);
      expect(mockValidationChange).toHaveBeenCalledWith(props.index,false);
    });
    it('with valid value',()=>{
      addInputContainer.find('input').simulate('change',{
        target:{value:'123'}
      });
      expect(mockAddingUserChange).toHaveBeenCalledWith(props.col.field,123);
      expect(mockValidationChange).toHaveBeenCalledWith(props.index,true);
    });
  });
});
