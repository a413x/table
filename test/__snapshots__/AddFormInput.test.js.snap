// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for AddFormInput component renders properly 1`] = `
<div>
  <label>
    #
  </label>
  <input
    className="form-control is-invalid"
    onChange={[Function]}
  />
  <div
    className="invalid-feedback"
  >
    Введите корректное значение.
  </div>
</div>
`;
